app/models/predicate_set.rb

```diff
@@ -39,12 +40,13 @@ class PredicateSet < ApplicationRecord
 
   belongs_to :strongest_match, class_name: "Predicate", optional: true
   has_one :configuration_profile
-  has_many :predicates
+  has_many :predicates, after_add: :update_max_weight, after_remove: :update_max_weight
 
   alias_attribute :name, :title
 
-  def max_weight
-    predicates.maximum(:weight)
+  def update_max_weight(_predicate)
+    self.max_weight = predicates.maximum(:weight)
```

Comment:
A simple `UPDATE` statement would be better since nothing else changes here:

```ruby
def update_max_weight(_predicate = nil)
  update_column(:max_weight, predicates.maximum(:weight))
end
```